packages/firestore/test/integration/api/aggregation.test.ts

```diff
@@ -18,24 +18,125 @@
+      const aggregateQuery_ = countQuery(query_);
+      expect(aggregateQuery_.query).to.be.equal(query_);
+    });
+  });
+
+  it('aggregateQuerySnapshot.query equals to aggregateQuery', () => {
+    const testDocs = {
+      a: { author: 'authorA', title: 'titleA' }
+    };
+    return withTestCollection(persistence, testDocs, async collection => {
+      const query_ = query(collection, where('author', '==', 'authorA'));
+      const aggregateQuery_ = countQuery(query_);
+      const snapshot = await getAggregateFromServerDirect(aggregateQuery_);
+      expect(snapshot.query).to.be.equal(aggregateQuery_);
+    });
+  });
+
+  it('aggregate query supports withConverter', () => {
+    const testDocs = {
+      a: { author: 'authorA', title: 'titleA' },
+      b: { author: 'authorB', title: 'titleB' }
+    };
+    return withTestCollection(persistence, testDocs, async collection => {
+      const query_ = query(
+        collection,
+        where('author', '==', 'authorA')
+      ).withConverter(postConverter);
       const countQuery_ = countQuery(query_);
       const snapshot = await getAggregateFromServerDirect(countQuery_);
-      expect(snapshot.getCount()).to.equal(3);
+      expect(snapshot.getCount()).to.equal(1);
     });
   });
+
+  it('aggregate query supports collection groups', () => {
+    return withTestDb(persistence, async db => {
+      const collectionGroupId = doc(collection(db, 'aggregateQueryTest')).id;
+      const docPaths = [
+        `${collectionGroupId}/cg-doc1`,
+        `abc/123/${collectionGroupId}/cg-doc2`,
+        `zzz${collectionGroupId}/cg-doc3`,
+        `abc/123/zzz${collectionGroupId}/cg-doc4`,
+        `abc/123/zzz/${collectionGroupId}`
+      ];
+      const batch = writeBatch(db);
+      for (const docPath of docPaths) {
+        batch.set(doc(db, docPath), { x: 1 });
+      }
+      await batch.commit();
+      const countQuery_ = countQuery(collectionGroup(db, collectionGroupId));
+      const snapshot = await getAggregateFromServerDirect(countQuery_);
+      expect(snapshot.getCount()).to.equal(2);
+    });
+  });
+
+  it('aggregate query fails if firestore is terminated', () => {
+    const testDocs = {
+      a: { author: 'authorA', title: 'titleA' }
+    };
+    return withTestCollection(
+      persistence,
+      testDocs,
+      async (collection, firestore) => {
+        await terminate(firestore);
+        const countQuery_ = countQuery(collection);
+        expect(() => getAggregateFromServerDirect(countQuery_)).to.throw(
+          'The client has already been terminated.'
+        );
+      }
+    );
+  });
+
+  it("terminate doesn't crash when there is flying aggregate query", () => {
+    const testDocs = {
+      a: { author: 'authorA', title: 'titleA' }
+    };
+    return withTestCollection(
+      persistence,
+      testDocs,
+      async (collection, firestore) => {
+        const countQuery_ = countQuery(collection);
+        getAggregateFromServerDirect(countQuery_).then();
```

Comment:
Can you try puting `void` at the beginning of this line, and removing `.then()`? That appears to be the "correct" way to ignore a promise.